models/sipnet/R/read_restart.SIPNET.R

```diff
@@ -19,59 +19,74 @@
 ##' @export
 read_restart.SIPNET <- function(outdir, runid, stop.time, settings, var.names, params) {
   
-  prior.sla <- params[[which(names(params) != "soil")[1]]]$SLA
+  prior.sla <- params[[which(!names(params) %in% c("soil", "soil_SDA", "restart"))[1]]]$SLA
   
   forecast <- list()
   
+  # additional varnames, because we need these deterministic relationships
+#  var.names <- c(var.names, "fine_root_carbon_content", "coarse_root_carbon_content")
+   
   # Read ensemble output
   ens <- read.output(runid = runid, 
                      outdir = file.path(outdir, runid), 
                      start.year = lubridate::year(stop.time), 
                      end.year = lubridate::year(stop.time),
                      variables = var.names)
   
-  last <- length(ens$NPP)
+  last <- length(ens$GWBI)
   
   forecast <- list()
   
-  # unit.conv <- (10000/1)*(1/1000)*(365.25*24*60*60) ## kgC m-2 s-1 -> MgC/ha/yr
+  kgms2Mghayr <- (10000/1)*(1/1000)*(365.25*24*60*60) ## kgC m-2 s-1 -> MgC ha-1 yr-1
```

Comment:
do you use these or only ud.convert?

Reply:
good point, these sort of conversions should definitely be superseded by udunits